File: backend/src/main/java/com/seismicrisk/model/NeighborhoodDefaults.java
package com.seismicrisk.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Map;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class NeighborhoodDefaults {
    private String neighborhoodId;
    private String neighborhoodName;
    private YearRange defaultYearRange;
    private Map<String, Double> typicalStructureTypes;
    private Double typicalNumFloorsMean;
    private Double typicalNumFloorsStd;
    private String typicalSoilClass;
    private Double retrofitRateEstimate;
    private String expectedMaterialQuality;
    
    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class YearRange {
        private int start;
        private int end;
    }
}

File: Programming/Personal/seismic-risk-app/lib/domain/entities/neighborhood_defaults.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'neighborhood_defaults.freezed.dart';
part 'neighborhood_defaults.g.dart';

@freezed
class NeighborhoodDefaults with _$NeighborhoodDefaults {
  const factory NeighborhoodDefaults({
    required String neighborhoodId,
    String? neighborhoodName,
    YearRange? defaultYearRange,
    Map<String, double>? typicalStructureTypes,
    double? typicalNumFloorsMean,
    double? typicalNumFloorsStd,
    String? typicalSoilClass,
    double? retrofitRateEstimate,
    String? expectedMaterialQuality,
  }) = _NeighborhoodDefaults;

  factory NeighborhoodDefaults.fromJson(Map<String, dynamic> json) =>
      _$NeighborhoodDefaultsFromJson(json);
}

@freezed
class YearRange with _$YearRange {
  const factory YearRange({
    required int start,
    required int end,
  }) = _YearRange;

  factory YearRange.fromJson(Map<String, dynamic> json) =>
      _$YearRangeFromJson(json);
}

